### PR TITLE
#1125 bug sms fragment

### PR DESCRIPTION
**Pull request by anagradova**
Corrected message formatting bugs in the template.py to match AWS formatting rules and updated test cases. All test case fragments were validated with https://messente.com/documentation/tools/sms-length-calculator

### CODE REVIEW

File: notifications_utils/template.py [hunk 1]
```diff
@@ -253,8 +254,35 @@ def fragment_count(self):
...
-        content_len = len(self.content_with_placeholders_filled_in.encode("utf8"))
-        return math.ceil(content_len / 140)
+
+        # check if all chars are in the GSM-7 character set
+        def gsm_check(x):
+            rule = re.compile(r'^[\sa-zA-Z0-9_@?£!1$"¥#è?¤é%ù&ì\\ò(Ç)*:Ø+;ÄäøÆ,<LÖlöæ\-=ÑñÅß.>ÜüåÉ/§à¡¿\']+$')
+            if not rule.search(x):
+                return False
+            else:
+                return True
```
Comment (ccostino) [inline notifications_utils/template.py:264]: Let's change this slightly just to make sure we're checking for the right thing.  According to [the Python re module documentation, `search()`](https://docs.python.org/3.9/library/re.html#re.search) returns `None` if no matches are found, which is a distinction between matching a pattern that just happens to be zero length.

In other words, the `None` return value is what we really want to test for to make sure the pattern wasn't found. 🙂 

We can also just test for the `None` return value and default to returning `True` to help simplify things a bit!

```suggestion
            gsm_match = rule.search(x)
            if gsm_match is None:
                return False
            return True
```

File: notifications_utils/template.py [hunk 2]
```diff
@@ -253,8 +254,35 @@ def fragment_count(self):
...
+
+        message_str = self.content_with_placeholders_filled_in
+
+        content_len = len(message_str)
+
+        """
+        Checks for GSM-7 char set, calculates msg size, and
+        then fragments based on multipart message rules. ASCII
+        was not specifically called out as almost all messages will
+        switch from 7bit GSM to Unicode.
```
Comment (ccostino) [inline notifications_utils/template.py:274]: These calculations are based on this calculator, correct? https://messente.com/documentation/tools/sms-length-calculator

If so, I'd just link to it here directly in the comment so we have a quick reference back to it in the future, e.g.:

```suggestion
        switch from 7bit GSM to Unicode.
        
        Calculations are based on https://messente.com/documentation/tools/sms-length-calculator
```

**Review by ccostino — COMMENTED** [review]
Thank you, @anagradova!

I have a couple of suggestions to make sure we're covering our bases properly, but overall this looks good to go to me. 🙂

**Comment by ccostino** [comment]
D'oh, sorry about the extra whitespace @anagradova!  I thought I had stripped it out when making the suggestions but it's hard to catch in the PR review UI. :-(

**Review by ccostino — APPROVED** [review]
Thank you again, @anagradova!
